tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/StridedSliceHelper.java

```diff
@@ -0,0 +1,214 @@
+
+      if (idx.beginMask()) {
+        beginMask |= 1L << i;
+      }
+
+      if (idx.endMask()) {
+        endMask |= 1L << i;
+      }
+
+      if (idx.ellipsisMask()) {
+        if(ellipsisMask != 0)
+          throw new IllegalArgumentException("Can not have two ellipsis in a slice");
+        ellipsisMask |= 1L << i;
+      }
+
+      if (idx.newAxisMask()) {
+        newAxisMask |= 1L << i;
+      }
+
+      if (idx.shrinkAxisMask()) {
+        shrinkAxisMask |= 1L << i;
+      }
+    }
+
+    return new StridedSliceArgs(begin, end, strides, beginMask, endMask, ellipsisMask, newAxisMask, shrinkAxisMask);
+  }
+  /**
+   * Return a strided slice from `input`.
+   *  <p>
+   *  The goal of this op is to produce a new tensor with a subset of
+   *  the elements from the `n` dimensional `input` tensor. The subset is chosen using
+   *  a sequence of `m` sparse range specifications encoded into the arguments
+   *  of this function. Note, in some cases
+   *  `m` could be equal to `n`, but this need not be the case. Each
+   *  range specification entry can be one of the following:
+   *  <p>
+   *  - An ellipsis (...) using {@link Indices#ellipsis()}. Ellipses are used to imply zero or more
+   *    dimensions of full-dimension selection and are produced using
+   *    `ellipsis_mask`. For example, `foo[...]` is the identity slice.
+   *  <p>
+   *  - A new axis using {@link Indices#newAxis()}. This is used to insert a new shape=1 dimension and is
+   *    produced using `new_axis_mask`. For example, `foo[:, ...]` where
+   *    `foo` is shape `(3, 4)` produces a `(1, 3, 4)` tensor.
+   *  <p>
+   *  - A range `begin:end:stride` using {@link Indices#slice(Long, Long, long)}  Index.slice()}. This is used to specify how much to choose from
+   *    a given dimension. `stride` can be any integer but 0.  `begin` is an integer
+   *    which represents the index of the first value to select while `end` represents
+   *    the index of the last value to select. The number of values selected in each
+   *    dimension is `end - begin` if `stride > 0` and `begin - end` if `stride < 0`.
+   *    `begin` and `end` can be negative where `-1` is the last element, `-2` is
+   *    the second to last. `begin_mask` controls whether to replace the explicitly
+   *    given `begin` with an implicit effective value of `0` if `stride > 0` and
+   *    `-1` if `stride < 0`. `end_mask` is analogous but produces the number
+   *    required to create the largest open interval. For example, given a shape
+   *    `(3,)` tensor `foo[:]`, the effective `begin` and `end` are `0` and `3`. Do
+   *    not assume this is equivalent to `foo[0:-1]` which has an effective `begin`
+   *    and `end` of `0` and `2`. Another example is `foo[-2::-1]` which reverses the
+   *    first dimension of a tensor while dropping the last two (in the original
+   *    order elements). For example `foo = [1,2,3,4]; foo[-2::-1]` is `[4,3]`.
```

Comment:
Documentation in non-generated classes should be converted from Python to Java, meaning that `foo[-2::-1]` and other examples like that must be converted to their Java equivalent.